gradient/cli/common.py

```diff
@@ -134,20 +140,62 @@ def generate_options_template(ctx, param, value):
         return value
 
     params = {}
+    objects_value_lists = {} # Object list name -> object key -> list of values
     for param in ctx.command.params:
         option_name = get_option_name(param.opts)
         if option_name in (OPTIONS_FILE_OPTION_NAME, OPTIONS_DUMP_FILE_OPTION_NAME):
             continue
 
         option_value = ctx.params.get(param.name) or param.default
 
+        # If this is an object list type option, add its value list to
+        # the specific object list set of value lists.
+        if isinstance(param, GradientObjectListOption):
+            new_value_list = option_value if option_value is not None else []
+            object_list_name = param.get_object_list_name()
+            if object_list_name not in objects_value_lists:
+                objects_value_lists[object_list_name] = {}
+            value_lists = objects_value_lists[object_list_name]
+            value_lists[param.get_object_key()] = new_value_list
+            continue
+
         if isinstance(param.type, cli_types.ChoiceType):
             for key, val in param.type.type_map.items():
                 if val == option_value:
                     option_value = key
 
         params[option_name] = option_value
 
+    # Transform value lists into objects
+    object_lists = {}
+    for object_list_name, value_lists in objects_value_lists.items():
+        # Find maximum length value list and assume it lines up
+        # with other object list options
+        if object_list_name not in object_lists:
+            object_lists[object_list_name] = []
+        object_list = object_lists[object_list_name]
+
+        num_items = 0
+        for object_key, value_list in value_lists.items():
+            if len(value_list) > num_items:
+                num_items = len(value_list)
```

Comment:
This can be shortened too using builtin max function and generator expression:
`num_items = max(len(value_list) for value_list in value_lists.values())`

[https://docs.python.org/3/library/functions.html#max](https://docs.python.org/3/library/functions.html#max)
[https://www.python.org/dev/peps/pep-0289/](https://www.python.org/dev/peps/pep-0289/)

Also notice you can use `.values()` on a dict if you only need to iterate over values (without keys), like in this example